### PR TITLE
add createReactContextHelpers()

### PR DESCRIPTION
**Pull request by VanTanev**
Enables the following pattern:

```typescript
// ExampleProvider.tsx
import React from 'react';
import { useInterpret } from '@xstate/react';
import { createReactContextHelpers } from 'xstate-helpers';
import { useErrorHandler } from 'react-error-boundary';
import { useAuth } from 'auth';
import { exampleMachine } from './example.machine';
export const {
  Provider: ExampleProvider,
  useInterpreter: useExampleInterpreter,
  useService: useExampleService,
  useSelector: useExampleSelector,
  useSend: useExampleSend,
} = createReactContextHelpers('Example', (props: { name: string }) => {
  const auth = useAuth();
  const handleError = useErrorHandler();
  const interpreter = useInterpret(exampleMachine, {
    context: { name: props.name },
    actions: {
      handleCriticalError: (_, e) => handleError(e.data),
    },
  });
  React.useEffect(() => {
    interpreter.send({ type: 'SET_USER', user: auth.user });
  }, [interpreter, auth.user]);
  return interpreter;
});
export default ExampleProvider;
```

```typescript
// App.tsx
import React from 'react';
import ExampleProvider, {
  useExampleInterpreter,
  useExampleService,
  useExampleSelector,
  useExampleSend,
} from './ExampleProvider';
const App: React.FC = () => {
  return (
    <ExampleProvider name="Example">
      <Component />
    </ExampleProvider>
  );
};
const Component: React.FC = () => {
  // the raw interpreter
  const interpreter = useExampleInterpreter();
  // just the send method, for components that don't need to read state
  const send = useExampleSend();
  // a pre-bound `useService()` hook for when you need the whole state
  const [state, send] = useExampleService();
  // a better pre-bound selector, that preserves proper types when React.useCallback() is used!
  // Favor using this over `useExampleService()` when possible, because selectors cause rerender
  // only when the selected value changes, while `useExampleService()` rerenders on every machine change.
  const name = useExampleSelector(
    React.useCallback(state => state.context.name, [])
  );
  // ...
};
```

@mattpocock @with-heart it would be great if you could give this a once-over before I publish it!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,72 @@ npm install xstate-helpers
 
 ## API Reference
 
+### createReactContextHelpers()
+
```
Comment (with-heart) [inline README.md:14]:
```suggestion

`createReactContextHelpers` creates a set of helpers that make it easy to share and access a machine through React context. 

Creates a `React.Context` which provides the machine's interpreter and returns a `Provider` for the context and `useInterpreter`, `useSend`, `useService`, and `useSelector` hooks which are initialized to the machine.

```

**Review by with-heart — APPROVED** [review]
So glad to have this in a package now!

**Comment by VanTanev** [comment]
I'm wondering if there is a better function name than `createReactContextHelpers()` :thinking:

**Comment by with-heart** [comment]
I was going to say `useMachineContextHelpers`, but remembered that machine also has `context` so that would be confusing. I feel like the only thing more specific would be `createReactContextMachineHelpers` or something, but that's getting a little verbose.

**Comment by with-heart** [comment]
Thanks again for discovering and building this pattern @VanTanev! Super useful!

**Pull request merged**